src/protocol/requests/fetch/v4/response.js

```diff
@@ -41,8 +41,16 @@ const decodeMessages = async decoder => {
     let records = []
 
     while (messagesDecoder.canReadBytes(RECORD_BATCH_OVERHEAD)) {
-      const recordBatch = await RecordBatchDecoder(messagesDecoder)
-      records = [...records, ...recordBatch.records]
+      try {
+        const recordBatch = await RecordBatchDecoder(messagesDecoder)
+        records = [...records, ...recordBatch.records]
+      } catch (e) {
+        // The tail of the record batches can have incomplete records
```

Comment:
Maybe throw in a reference to https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-FetchAPI and the passage "As an optimization the server is allowed to return a partial message at the end of the message set. Clients should handle this case.", since "due to how maxBytes works" doesn't really tell you much unless you already know what it means.